superset/utils/core.py

```diff
@@ -1764,6 +1765,31 @@ def apply_max_row_limit(
     return max_limit
 
 
+def apply_max_row_limit_table(
```

Comment:
Same as with `_process_row_limit`, why not extending the current one instead of creating a table specific to handle optional param?